### DISCUSSION TITLE
How to change the method name creation for a NSwag.MSBuild generated client

### DISCUSSION THREAD

**Discussion by xDDDx** [RicoSuter/NSwag#4772] [Q&A]
Hi, 

I have an openapi 3 defnition, that contains routes like this '/$batch'.  I use NSwag.MSBuild to generate the client. The generated method names in a C# client are not valid, so the client is broken. Is it possible to configure NSwag some how to generate valid method names or to do something else to fix this issue. 

I don't have the possibility to change the openapi definition. We are getting this from outside.

Thanks a lot

**Comment by drwharris** [comment]
This might help https://github.com/RicoSuter/NSwag/wiki/CSharpClientGenerator

Im guessing you can implement you own https://github.com/RicoSuter/NSwag/blob/master/src/NSwag.CodeGeneration.CSharp/CSharpClientGenerator.cs

**Comment by xDDDx** [comment]
The problem is, I use it as a pre build approach inside the .csproj
`<Target Name="NSwag" BeforeTargets="Build">...`

So I don't have code behind. It is more like NSwagStudio generation. The question is, is it neverless possible to use some kind of custom ClientGenerator, or do I need to switch to client generation from code?

**Comment by xDDDx** [comment]
Is it maybe better to change the method name generation? Like it is for parameters?
```
var variableName = ConversionUtilities.ConvertToLowerCamelCase(name
                    .Replace("-", "_")
                    .Replace(".", "_")
                    .Replace("$", string.Empty)
                    .Replace("@", string.Empty)
                    .Replace("[", string.Empty)
                    .Replace("]", string.Empty), true);
```
